### PR TITLE
Add React custom hook post

### PR DESCRIPTION
**Pull request by regexpressyourself**
A post on easily extracting custom hooks in React.

### CODE REVIEW

File: _drafts/2022-01-25-quick-and-easy-custom-hooks.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+Notice very little has changed. This works functionally the same as before, but now we can call `useQueryParam` from multiple different components without needing to repeat code.
+
+
+
+## tl;dr
+
+To extract a custom hook from a complex component:
+1. Take all the logic above your component's JSX, and move it to a custom hook. 
+2. Return any data needed in your component from your custom hook.
+3. Call the custom hook from your component, and pull out whatever data is needed.
```
Comment (bwright77) [inline _drafts/2022-01-25-quick-and-easy-custom-hooks.md:180]: The hook vs component repetition here is making it hard to understand quickly. May want number 2 to just say 'Return any data needed to render the JSX from your custom hook` but then 3 would be repetitive. Anyway, just think there's some repetition clean-up we could do here for clarity.

**Review by bwright77 — APPROVED** [review]
Really informative! Love it!

File: _drafts/2022-01-25-quick-and-easy-custom-hooks.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+Essentially, hooks are functions that run on each render of a component. They are semantically differentiated from normal functions by using the `use` keyword at the front of the hook name (e.g. `useState`). Hooks can take parameters and can return values, but neither is required.
+
+## Why use a React hook?
+
+### Manage complexity
+
+We can compose custom hooks by leveraging other hooks like `useEffect` and `useState`. This allows us to write custom hooks that can monitor pieces of application state and send information back to our components to handle the UI when things change.
+
+While all this can be done at the component level, using custom hooks allows us to easier separate the heavy logic from the UI.
```
[the remark this answers is not in the document]
Comment (bwright77) [reply _drafts/2022-01-25-quick-and-easy-custom-hooks.md:30]: maybe `using custom hooks makes it easier to separate the heavy logic`...

**Review by eallegrucci — APPROVED** [review]
Awesome article! I just learned something new :)

File: _drafts/2022-01-25-quick-and-easy-custom-hooks.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (regexpressyourself) [reply _drafts/2022-01-25-quick-and-easy-custom-hooks.md]: I like it. Definitely a weird tense.

File: _drafts/2022-01-25-quick-and-easy-custom-hooks.md (hunk 1 above)
Comment (regexpressyourself) [reply _drafts/2022-01-25-quick-and-easy-custom-hooks.md]: That's a good point. I don't think 3 is necessarily repetitive after rewording 2. Curious what you think of the new wording?

https://github.com/himaxwell/himaxwell.github.io/pull/19/commits/9e18f83255c69c1ea146dc7184f3ba8931672d0d

**Review by CaseyKelly — APPROVED** [review]
Great post, Sam! Everything looks correct to me 🕵️  💯

**Comment by regexpressyourself** [comment]
Thanks all!

**Comment by dpfeil** [comment]
Already merged, but also wanted to say thanks, @regexpressyourself!

**Pull request merged**